scipy/integrate/_ode.py

```diff
@@ -912,8 +912,8 @@ def run(self, f, jac, y0, t0, t1, f_params, jac_params):
         y1, t, istate = self.runner(*args)
         if istate < 0:
             warnings.warn(self.__class__.__name__ + ': ' +
-                          self.messages.get(istate,
-                                            'Unexpected istate=%s' % istate))
+                          self.messages.get(istate) +
+                          ('Unexpected istate=%s' % istate)))
```

Comment:
This is not correct (will fail when istate is unexpected); the original code is correct.
The original intent is to show the "Unexpected istate" message only when the istate is unknown i.e. not one of those in the `messages` list.

In principle, it is possible to change the message to always show the istate, but you need to consider what happens if istate is unknown.

Same issue applies to changes below.

Also, now that you are changing it, minor nitpicking: it is stylistically somewhat better to use `{} {} {} {}`.format(...) for string formatting rather than concatenating parts with `+`.